staging/src/k8s.io/apiserver/pkg/registry/rest/rest.go

```diff
@@ -207,6 +207,24 @@ type UpdatedObjectInfo interface {
 	UpdatedObject(ctx context.Context, oldObj runtime.Object) (newObj runtime.Object, err error)
 }
 
+// UpdateConfig provides information about the validation functions that
+// the Updater should use, and allows for overriding certain UpdateStrategy behavior.
+type UpdateConfig interface {
```

Comment:
I'm still really not sold on the interface here. Why would you have multiple versions of this? It's really just a POD.

Reply:
I like enforcing the construction of it to specify every option. the downside of a struct with public fields is that a new field doesn't force call sites to define that field. zero values are sometimes reasonable, but not always

Reply:
Sounds good